src/service.rs

```diff
@@ -57,7 +57,8 @@ impl GrpcService {
     ) -> Result<GrpcResult, StatusCode> {
         let failure_mode = operation.get_failure_mode();
         if let Some(res_body_bytes) =
-            hostcalls::get_buffer(BufferType::GrpcReceiveBuffer, 0, resp_size).unwrap()
+            hostcalls::get_buffer(BufferType::GrpcReceiveBuffer, 0, resp_size)
+                .expect("failed to get_buffer on GrpcReceiveBuffer")
```

Comment:
Should this leverage `FailureMode`?

Reply:
Yep good point, this and below should be handled